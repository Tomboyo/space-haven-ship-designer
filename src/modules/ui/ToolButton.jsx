export default function ToolButton({ active, children, onClick }) {
  const className = active ? "active" : "";
  return (
    <button className={className} onClick={onClick}>
      {children}
    </button>
  );
}
