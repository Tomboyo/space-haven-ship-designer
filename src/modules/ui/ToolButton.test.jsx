import ToolButton from "./ToolButton.jsx";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";

it("Uses active styling when active", () => {
  const component = renderer.create(
    <ToolButton active={true} onClick={() => {}} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Does not use active styling when inactive", () => {
  const component = renderer.create(
    <ToolButton active={false} onClick={() => {}} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Triggers onClick callback when clicked", () => {
  const mockOnClick = jest.fn();
  const { queryByText, getByText } = render(
    <ToolButton active={false} onClick={mockOnClick}>
      Label
    </ToolButton>,
  );

  expect(queryByText(/Label/i)).toBeTruthy();
  fireEvent.click(getByText(/Label/i));

  expect(mockOnClick.mock.instances.length).toBe(1);
});
