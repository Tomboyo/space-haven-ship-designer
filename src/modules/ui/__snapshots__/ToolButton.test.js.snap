// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does not use active styling when inactive 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;

exports[`Uses active styling when active 1`] = `
<button
  className="active"
  onClick={[Function]}
/>
`;
